[MDCTextfields] Added accessibility label tests for MDCMultilineTextFields

randallli opened the pull request:
Copied the logic from the single line textfield. over to the multiline one.

part of https://github.com/material-components/material-components-ios/issues/7513

randallli committed:
[TextField] Added tests for accessibilityLabel

randallli committed:
feedback from PR

randallli committed:
better a11y strings for readability

randallli committed:
clang format

randallli committed:
corrected wrong string

randallli committed:
Merge commit '3d1b6f9acc0eeea2f5a2ae820d7c6ede437df628' into tfTest

randallli committed:
[MDCTextfields] Added accessibilty label tests to MDCMultilineTextFields

jverkoey commented on components/TextFields/tests/unit/MDCMultilineTextFieldAccessibilityTests.m:
```diff
@@ -0,0 +1,91 @@
+  XCTAssertEqualObjects(view.accessibilityLabel, @"main accessibility label");
+}
+
+- (void)testMDCMultilineTextFieldaccessibilityLabel {
```
You can likely drop `MDCMultilineTextField` from all of these test names because it's implicit in the class name.

randallli replied:
Done.

material-automation commented:
bazel detected changes to the following targets:

```
//components/Chips:ChipThemer
//components/Chips:Chips
//components/Chips:ColorThemer
//components/Chips:FontThemer
//components/Chips:ObjcExamples
//components/Chips:ShapeThemer
//components/Chips:SwiftExamples
//components/Chips:Theming
//components/Chips:TypographyThemer
//components/Chips:privateHeaders
//components/Chips:snapshot_test_lib
//components/Chips:snapshot_tests
//components/Chips:snapshot_tests.swift_runtime_linkopts
//components/Chips:snapshot_tests_test_binary
//components/Chips:snapshot_tests_test_bundle
//components/Chips:unit_test_sources
//components/Chips:unit_test_swift_sources
//components/Chips:unit_tests
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3.swift_runtime_linkopts
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3_test_binary
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3.swift_runtime_linkopts
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3_test_binary
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Chips:unit_tests_environment
//components/Chips:unit_tests_environment.swift_runtime_linkopts
//components/Chips:unit_tests_environment_test_binary
//components/Chips:unit_tests_environment_test_bundle
//components/NavigationBar:ObjcExamples
//components/TextFields:ColorThemer
//components/TextFields:FontThemer
//components/TextFields:ObjcExamples
//components/TextFields:SwiftExamples
//components/TextFields:TextFields
//components/TextFields:Theming
//components/TextFields:TypographyThemer
//components/TextFields:snapshot_test_lib
//components/TextFields:snapshot_tests
//components/TextFields:snapshot_tests.swift_runtime_linkopts
//components/TextFields:snapshot_tests_test_binary
//components/TextFields:snapshot_tests_test_bundle
//components/TextFields:unit_test_sources
//components/TextFields:unit_test_swift_sources
//components/TextFields:unit_tests
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3.swift_runtime_linkopts
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3_test_binary
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3.swift_runtime_linkopts
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3_test_binary
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/TextFields:unit_tests_IPHONE_X_IN_11_0
//components/TextFields:unit_tests_IPHONE_X_IN_11_0.swift_runtime_linkopts
//components/TextFields:unit_tests_IPHONE_X_IN_11_0_test_binary
//components/TextFields:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/schemes/Shape:ObjcExamples
```

<!-- identifier: affected-targets -->

randallli committed:
added logic needed

randallli commented:
@jverkoey I added significant logic that I think a re-review is warranted.
In my first pass I thought that the tests were passing because the logic was already there. I have since realized that it was not in the multiline textfield so I adjusted it to have it.

randallli committed:
Merge branch 'develop' into tfTest

jverkoey commented on components/TextFields/src/MDCMultilineTextField.m:
```diff
@@ -782,6 +782,23 @@ - (NSString *)accessibilityValue {
 
+- (NSString *)accessibilityLabel {
+  NSMutableArray *accessibilityStrings = [[NSMutableArray alloc] init];
+  if ([super accessibilityLabel].length > 0) {
```
Consider storing `[super accessibilityLabel]` in a local variable so that we're not calling super multiple times here.

randallli replied:
We are only calling it 2 times. I dont think thats needed.

jverkoey replied:
This is not about performance, it's about possible unintended side effects from calling super multiple times. In general we should not be calling super multiple times within a subclass overridden API.

jverkoey replied:
Please fix in a follow-up PR.

jverkoey replied:
Apologies my statement was not as clear about the desired change in the original comment.
